### ISSUE TITLE
Can't support Windows 98?

### ISSUE THREAD

**Issue by denjhang** [necroware/ps2-serial-mouse-adapter#6]
I connected the adapter to the serial port of my 586 motherboard according to the correct method, and then turned it on again. I found that although Windows 98 can recognize the mouse normally, when I move the mouse, there is no response at all, and there is no response when I click the left and right buttons of the mouse. Only when I press the reset button of the arduino, there is a small probability that the mouse will move for a moment.

**Comment by denjhang** [comment]
![EA7DD5F0EC61F3DD1058ED9C498ED7FC](https://user-images.githubusercontent.com/61811583/147396720-811f83c6-3f5f-44a9-b621-af8dc3ebe36d.png)
![2B4B0AB32CF32BE652F5EFF7C4E6259E](https://user-images.githubusercontent.com/61811583/147396721-9ac41fb7-d516-4975-89e0-00c580d3c191.png)

![1B9491AA2935EC1BB71709A583C2417B](https://user-images.githubusercontent.com/61811583/147396723-feeca2ae-6d53-4d73-bde5-71512e375912.png)

**Comment by denjhang** [comment]
Of course, this mouse is a bargain without a brand, but I don't think this is a problem. Because the mouse can be used normally on MSX with another converter.

**Comment by necroware** [comment]
Which software version did you use? If you used the one from the release, would you please try the HEAD from main? There is a fix for streaming mode bug (see #2)

Furthermore some mice seem to have problems with the unstable clock. You can try to add a 10k or less resistor between the clock pin (5) and the VCC pin (21) of the Arduino. See the picture in #4. If this is the problem with your mouse, the fixed PCB design is already waiting for the merge.

**Comment by creopard** [comment]
In general, the adapter works flawlessly with Windows 98 (as an example Windows 98 SE and a Microsoft IntelliMouse Optical):

![comport1](https://user-images.githubusercontent.com/14175483/147405347-64c1f0ba-50ee-4a23-88f8-be4441104671.jpg)
![win98](https://user-images.githubusercontent.com/14175483/147405345-cdae71ad-c3eb-43ba-bfd3-768d372ade58.jpg)

You could also try to check your IRQ settings.
Is your "autoexec.bat" free from any "ctmouse.exe" entries?

**Comment by denjhang** [comment]
I found out that my "autoexec.bat" is empty and there is nothing in it.

**Comment by necroware** [comment]
That's good. Then try the things I wrote previously. The latest software from the main brunch and/or the pull-up resistor on the clock pin.

**Comment by denjhang** [comment]
I tried to connect a 10K resistor, and found that after the mouse started in Windows 98, it quickly moved to the lower left corner of the screen and kept pressing the right button of the mouse.

**Comment by necroware** [comment]
Ok, that's a progress. As I said, some mice seem to have problems with the clock stability. If you have other resistor values at hand it would be nice if you can try them. something between 20kOhm and 4,7kOhm. I would really like to know, which one works better. So far we found only one mouse which had some issues and we could solve them by using a 10kOhm resistor.

I think this issue here is a duplicate of #4 so far. Can you please also try to change the code in one place and see if something changes? You have to change the line 89 in the file Ps2Mouse.cpp:

from: pinMode(m_ref.m_clockPin, INPUT_PULLUP);
into: pinMode(m_ref.m_clockPin, INPUT);

**Comment by denjhang** [comment]
So far I have tried 10K and 20K, and there is no difference. As for modifying the code, I don't have a USB-TTL cable at the moment, so I should be able to test it in a few days. (Actually, this converter was made for me by other people, I am going to make one myself)

**Comment by necroware** [comment]
Ok, thank you for testing. Since you can't test the latest software version, could you please set the JP1 3-4 jumper? That would enforce streaming mode and if your mouse is working with that, then it is affected by the bug #3.

Seems like the adapter has a problem with your mouse, but with your help I think we will be able to find out what's wrong. I hope, the USB-TTL shouldn't be a problem to find?
